Shop-1/src/controllers/products.ts

```diff
@@ -0,0 +1,110 @@
+import Product from "../models/Products";
+import Product from "../models/Product";
+import { reqRes, reqResNext } from "../utils/interfaces";
+import { ProductsNotFoundError } from "../utils/errors";
+
+// CREATE
+
+export const postProduct: reqRes = async (req, res) => {
+  try {
+    const product = new Products(req.body);
+    await product.save();
+    res.status(201).send(product);
+  } catch (err) {
+    res.status(400).send("Server Error");
+  }
+};
+
+// READ ALL w/ PARAMS
+
+export const getProductParams: reqResNext = async (req, res, next) => {
+  try {
+    if (Object.keys(req.query).length <= 0) next();
+    else {
+      let Products = await Products.find();
```

Comment:
let products = await Products.find();

declare "let products" with lowercase P. Products with cap is your product model.

Reply:
then update all the following products to be lowercase p when you are dealing with the mongo documents.